docs/guides/upgrading-v3.md

```diff
@@ -213,28 +223,55 @@ That allowed us to make the KMS configuration consistent with all other AWS reso
+By default, Lambda version hashes will now be generated using a more robust algorithm (fixes determinism issues). Since this change requires manual effort during the migration, you can keep using the old algorithm in v3 via the following configuration:
 
-```
+```yaml
 provider:
   lambdaHashingVersion: 20200924
 ```
 
-However, we highly encourage an upgrade to the new algorithm. To upgrade, you must redeploy your service with code or configuration change in all functions. You can do it by following the guide below:
+Adding the above configuration is sufficient to be compatible with v3.
+
+However, we highly encourage upgrading to the new algorithm. To upgrade, you must:
+
+1. Enable the new hashing mechanism in v2 by setting:
+
+   ```yaml
+   provider:
+     lambdaHashingVersion: 20201221
+   ```
 
-**NOTE**: Please keep in mind that these changes require two deployments with manual configuration adjustment between them. It also creates two additional versions and temporarily overrides descriptions of your functions. Migration will need to be done separately for each of your environments/stages.
+2. Redeploy **with code or configuration changes** in all functions.
 
-1. Run `sls deploy` with additional `--enforce-hash-update` flag: that flag will override the description for Lambda functions, which will force the creation of new versions.
-2. Set `provider.lambdaHashingVersion` to `20201221` in your configuration: your service will now always deploy with the new Lambda version hashes (which is the new defualt in v3).
-3. Run `sls deploy`, this time without additional `--enforce-hash-update` flag: that will restore the original descriptions on all Lambda functions.
+   If your deployment doesn't contain changes, you will get the following error: "_A version for this Lambda function exists._"
 
-Now your whole service is fully migrated to the new Lambda Hashing Algorithm.
+   To force changes in all functions, you can deploy code changes, upgrade dependencies, or even temporarily create empty files in your codebase.
 
-If you do not want to temporarily override descriptions of your functions or would like to avoid creating unnecessary versions of your functions, you might want to use one of the following approaches:
+   Alternatively, you can use the `--enforce-hash-update` helper:
+
+   1. Run `serverless deploy` with the `--enforce-hash-update` flag: that flag will force changes by temporarily overriding the Lambda function descriptions (there is no runtime impact).
+   2. Run `serverless deploy` without the flag above to restore the descriptions on all Lambda functions.
+
+Remember that you will need to deploy with changes to each stage you have previously deployed. For development stages another option is to remove and recreate the stage entirely.
```

Comment:
As discussed on Slack, I changed from:

1. `sls deploy --enforce-hash-update`
2. upgrade config
3. `sls deploy`

to:

1. upgrade config
2. `sls deploy --enforce-hash-update`
3. `sls deploy`

Benefits:

- slightly easier to understand that we only create the fake deployment to force changes
- the time between the 2 deploys is shorter/more controllable, which may make some people more comfortable with "messing up the descriptions" in prod (for example)
- easier to do across stages: set the flag, then deploy to all stages

Also it allowed me to merge all different options into a single list, making the thing (hopefully) simpler to learn.